app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -78,15 +80,20 @@ def import_for_repository(repository, force = false)
           ids_to_import = ids
         end
         self.import_all(ids_to_import, :index_repository_association => false) if repository.content_view.default? || force
-        self.sync_repository_associations(repository, ids) if self.manage_repository_association
+        self.sync_repository_associations(repository, :uuids => ids) if self.manage_repository_association
       end
 
       def unit_id_field
         "#{self.name.demodulize.underscore}_id"
       end
 
-      def sync_repository_associations(repository, unit_uuids, additive = false)
-        associated_ids = with_uuid(unit_uuids).pluck(:id)
+      def sync_repository_associations(repository, options = {})
+        additive = options.fetch(:additive, false)
+        associated_ids = options.fetch(:ids, nil)
+        uuids = options.fetch(:uuids) if associated_ids.nil?
+
+        associated_ids = with_uuid(uuids).pluck(:id) if uuids
+
```

Comment:
how about
```
associated_ids = if options.key?(:ids)
                    options[:ids]
                 else
                    uuids = options.fetch(:uuids, [])
                    with_uuid(uuids).pluck(:id)
                 end
```

Reply:
I think it's a fine way to do it, but if I wrote it, maybe I'd do something like this:

```
associated_ids = options.fetch(
  :ids,
  with_uuid(options.fetch(:uuids)).pluck(:id)
)
```

I assume you want it to fail if uuids was not provided (i.e. KeyError in fetch)?

Reply:
@parthaa `with_uuid(options.fetch(:uuids, [])).pluck(:id)` would result in `NoMethodError: undefined method 'pluck' for Array`. I think we'd rather have the `KeyError` of `options.fetch(:uuids)`.

Reply:
yeah the goal was to fail if either :ids or :uuids are not provided.    I tend to prefer that when methods use a hash of options, the fetch's be at the top so its easy to see what can be passed in.  Its usually a bit more verbose, but most of the suggestions make it harder to see those options.  Because of that, I tend to lean towards leaving it the way it is, but am open to more discussion